docs/fundamentals/annotations-overview.md

```diff
@@ -0,0 +1,265 @@
+    where T : class, IResource
+{
+    return builder
+        .WithUrl("metrics", "/metrics")
+        .WithUrl("health", "/health");
+}
+```
+
+### Annotation validation
+
+Validate annotations to ensure they're correctly configured:
+
+:::code source="snippets/annotations-overview/Program.cs" id="ValidatedAnnotation":::
+
+## Accessing annotations
+
+You can access annotations on resources for inspection or processing:
+
+:::code source="snippets/annotations-overview/Program.cs" id="AccessingAnnotations":::
+
+## Testing with annotations
+
+When writing tests, you can inspect annotations to verify resource configuration:
+
+```csharp
+[Fact]
+public async Task Resource_Should_Have_Expected_Annotations()
+{
+    var appHost = await DistributedApplicationTestingBuilder
+        .CreateAsync<Projects.MyApp_AppHost>();
+
+    await using var app = await appHost.BuildAsync();
+
+    var resource = app.Resources.GetResource("my-resource");
+
+    // Assert that specific annotations exist
+    Assert.NotEmpty(resource.Annotations.OfType<CustomConfigAnnotation>());
+
+    // Assert annotation properties
+    var config = resource.Annotations
+        .OfType<CustomConfigAnnotation>()
+        .First();
+    
+    Assert.Equal("expected-value", config.ConfigValue);
+}
+```
+
+## Best practices
+
+When working with annotations, consider these best practices:
+
+### Use meaningful names
+
+Choose descriptive names for your annotation classes and properties:
+
+```csharp
+// Good
+public sealed class DatabaseConnectionPoolAnnotation : IResourceAnnotation
+
+// Avoid
+public sealed class DbAnnotation : IResourceAnnotation
+```
+
+### Follow the builder pattern
+
+Create fluent extension methods that follow .NET Aspire's builder pattern:
+
+```csharp
+var resource = builder.AddMyResource("name")
+    .WithCustomBehavior()
+    .WithAnotherFeature();
+```
+
+### Document your annotations
+
+Provide XML documentation for custom annotations and extension methods:
+
+```csharp
+/// <summary>
+/// Configures custom caching behavior for the resource.
+/// </summary>
+/// <param name="builder">The resource builder.</param>
+/// <param name="ttl">The time-to-live for cached items.</param>
+/// <returns>The resource builder for chaining.</returns>
+public static IResourceBuilder<T> WithCaching<T>(
+    this IResourceBuilder<T> builder,
+    TimeSpan ttl)
+    where T : class, IResource
+```
+
+### Keep annotations simple
+
+Annotations should be focused on a single responsibility:
+
+```csharp
+// Good - single responsibility
+public sealed class RetryPolicyAnnotation : IResourceAnnotation
+{
+    public int MaxRetries { get; set; }
+    public TimeSpan Delay { get; set; }
+}
+
+// Avoid - multiple responsibilities
+public sealed class ConfigAnnotation : IResourceAnnotation
+{
+    public RetryPolicy RetryPolicy { get; set; }
+    public LoggingSettings Logging { get; set; }
+    public SecuritySettings Security { get; set; }
+}
+```
```

Comment:
These sections could be improved to better communicate how custom annotations are intended to be authored and used. They're specific resources within an app host, often being leveraged as part of other extension methods to infer metadata or user intent, or to reason about how a resource. Likewise, they're used within lifecycle hooks to query the app model for the same reasons.